Introduce new ZInfoNTPSource message in `proto/info/ntpsource.proto`

rouming opened the pull request:
`ZInfoNTPSource` message contains a list of peers describing to which NTP daemon (e.g. chronyd) is synchronized on EVE side. This message will be sent periodically from EVE to the `/info` controller endpoint.

CC: @zed-rishabh 
CC: @eriknordmark

rouming commented:
There is one yetus complain 
```
buflint:Enum zero value name "NTP_PEER_STATE_SYNC" should be suffixed with "_UNSPECIFIED".
```
which I can't fix, because 0 state means "SYNC" and not UNSPECIFIED.

eriknordmark commented on proto/info/ntpsources.proto:
```diff
@@ -0,0 +1,74 @@
+
+// Enum for describing the NTP source state
+enum NTPSourceState {
+  NTP_SOURCE_STATE_SYNC        = 0;
```
This buflint should be fixed since zero needs to be reserved (by definiting an _UNSPECIFIED name) and the real values starting at 1.

eriknordmark commented on proto/info/ntpsources.proto:
```diff
@@ -0,0 +1,74 @@
+
+// Enum for describing the NTP source mode
+enum NTPSourceMode {
+  NTP_SOURCE_MODE_CLIENT   = 0;
```
Ditto

eriknordmark commented:
> There is one yetus complain
> 
> ```
> buflint:Enum zero value name "NTP_PEER_STATE_SYNC" should be suffixed with "_UNSPECIFIED".
> ```
> 
> which I can't fix, because 0 state means "SYNC" and not UNSPECIFIED.

Is NTPSourceState and NTPSourceMode from some external/standard document? If so it would be good to put a reference to such a document.
And add whatever comment buflint needs to disable the check for that line. See https://buf.build/docs/configuration/v1/buf-yaml#allow_comment_ignores

rouming commented:
> > which I can't fix, because 0 state means "SYNC" and not UNSPECIFIED.
> 
> Is NTPSourceState and NTPSourceMode from some external/standard document? If so it would be good to put a reference to such a document. And add whatever comment buflint needs to disable the check for that line. 

mode and states are from the chrony itself. I added explanation and link to the sources.

I fixed the yetus for _UNSPECIFIED part by keeping state and mode +1, but there are more not related to that. I do not want to spoil each line with some yetus-disabling comment. This makes code looks ugly and the overall impression is that not a tool for development, but development for the tool making it happy. As usual 90% useless.

rouming commented:
@eriknordmark could you please take a look and approve/merge if no objections. Want this changes to be taken by the other teams. Thanks.

eriknordmark reviewed (approved):
LGTM

zed-rishabh commented on proto/info/ntpsources.proto:
```diff
@@ -0,0 +1,80 @@
+}
+
+// As described in http://doc.ntp.org/current-stable/ntpq.html
+message NTPSource {
+  bool     authenticated             = 1;
+  bool     reachable                 = 2; // (reachability == 0xff), i.e. 8 attempts
+  uint32   reachability              = 3;
+  NTPSourceState state               = 4;
+  NTPSourceMode mode                 = 5;
+  string   hostname                  = 6;
+  string   src_addr                  = 7;
+  uint32   src_port                  = 8;
+  string   dst_addr                  = 9;
+  uint32   dst_port                  = 10;
+  uint32   leap                      = 11;
+  uint32   stratum                   = 12;
+  uint32   precision                 = 13;
+  string   ref_id                    = 14;
+  google.protobuf.Timestamp ref_time = 15;
+  sint32   poll                      = 16;
+  uint32   flags                     = 17; // bitmap from NTPSourceBitmap
+  double   offset                    = 18;
+  double   delay                     = 19;
+  double   dispersion                = 20;
+  double   jitter                    = 21;
+  double   root_delay                = 22;
+  double   root_disp                 = 23;
+}
```
What will be the unique identifier?
